### ISSUE TITLE
Make <feature>RuntimeElements extend <feature>RuntimeOnly

### ISSUE THREAD

**Issue by jjohannes** [gradle/gradle#13278]
This is missing during the _configurations_ setup [here](https://github.com/gradle/gradle/blob/30cbbf848e0e1c4e3df6117e4bcbeea74e713f90/subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/DefaultJavaFeatureSpec.java#L121).

The current behavior is inconsistent with what we do for the _main_ feature of a Java Library, where `runtimeElements` extends `runtimeOnly`.

As _test fixtures_ use _feature variants_ under the hood, this also concerns the relationship between `testFixturesRuntimeElements` and `testFixturesRuntimeOnly`.

I see this as a bug, and thus think we can change this in a minor with a note in the "potentially breaking changes" section of the upgrading guide. The `JavaTestFixturesPlugin` is still incubating. We can also wait until 7.0 to fix this.

The workaround is to introduce the relationship yourself in your build file or a custom plugin:
```
testFixturesRuntimeElements.extendsFrom(testFixturesRuntimeOnly)
```

**Comment by melix** [comment]
Correct me if I'm wrong, but the runtime elements shouldn't expose `runtimeOnly` and the api elements shouldn't expose `compileOnly`. I think you meant _classpath_, not _elements_.

**Comment by jjohannes** [comment]
No I meant `runtimeElements`. Runtime always needs to be transitive. I need all my classes visible at runtime to run my software. (See slo Java Library Plugin documentation.)

I think runtime classpath gets already wired correctly.

Note:
For `compileOnly` we made the decision to not expose (it is  effectively `compileOnlyImplementation`). But we could have `compileOnlyApi` as well, although the use cases are more limited for that (annotation libraries you need at compile time on the API but not at runtime would be a use case for that).

**Comment by melix** [comment]
Right!

**Comment by ljacomet** [comment]
This has been fixed through #13650
